.github/workflows/tdd.yml

```diff
@@ -0,0 +1,94 @@
+name: Jest Tests TDD
+
+on:
+  push:
+    branches: [ "main", "feat/**" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  jest_tests:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up directories
+        run: |
+          mkdir tests sources sources/frontend sources/backend sources/db
+
+      - name: "[Database] Checkout code"
+        run: |
+          git clone https://github.com/DevOps-2023-TeamA/tsao-db.git sources/db
+        working-directory: ${{ github.workspace }}
+
+      - name: "[Database] Setup MySQL"
+        uses: mirromutth/mysql-action@v1.1
+        with:
+          host port: 3306
+          container port: 3306
+          mysql root password: root
+          mysql database: user
+          mysql user: root 
+          mysql password: root
+
+      - name: "[Database] Wait for MySQL"
+        run: sleep 10
+
+      - name: "[Database] Create Tables"
+        run: |
+          mysql --protocol=TCP -h localhost -P 3306 --user=root --password=root user < sources/db/create_table.sql
+
+      - name: "[Database] Insert Sample Data"
+        run: |
+          mysql --protocol=TCP -h localhost -P 3306 --user=root --password=root user < sources/db/insert_data.sql
+
+      - name: "[Backend] Checkout code"
+        run: |
+          git clone https://github.com/DevOps-2023-TeamA/tsao-backend-svc.git sources/backend
+        working-directory: ${{ github.workspace }}
+
+      - name: "[Backend] Setup Go environment"
+        uses: actions/setup-go@v5.0.0
+
+      - name: "[Backend] Deploy Auth Microservice"
+        run: |
+          cd sources/backend
+          go run microservices/auth/auth.go -sql "root:root@protocol@tcp(127.0.0.1:3306)/tsao" &
+        working-directory: ${{ github.workspace }}
+
+      - name: "[Backend] Deploy Users Microservice"
+        run: |
+          cd sources/backend
+          go run microservices/users/users.go -sql "root:root@protocol@tcp(127.0.0.1:3306)/tsao" &
+        working-directory: ${{ github.workspace }}
+
+      - name: "[Backend] Deploy Records Microservice"
+        run: |
+          cd sources/backend
+          go run microservices/records/records.go -sql "root:root@protocol@tcp(127.0.0.1:3306)/tsao" &
+        working-directory: ${{ github.workspace }}
+
+      - name: "[Frontend] Checkout code"
+        uses: actions/checkout@v2
+        with:
+          path: sources/frontend
+
+      - name: "[Testing] Setup Node.js environment"
+        uses: actions/setup-node@v4.0.1
+
+      - name: "[Testing] Install Jest"
+        run: |
+          npm install 
+        working-directory: ${{ github.workspace }}
+
+      - name: "[Testing] Run Jest Test Cases"
```

Comment:
these Jest scripts might work best on the tests.yml project
Especially since the deployment code got modified in the last bit, just hold for a bit